### ISSUE TITLE
Rails/Date should not check for .to_time

### ISSUE THREAD

**Issue by btoconnor** [bbatsov/rubocop#5931]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior
With rubocop, I expect to be able to granularly turn on separate linters based on the needs of the repository I am working on.

With `Rails/Date`, I expect it to flag one particular issue: the usage of `Date.today` (or similar variant from `Date`).

## Actual behavior
Currently, the `Rails/Date` linter checks for two separate things:
1. It checks to see if you're using `Date.today` (or some similar variant), and flags an issue telling you to use `Time.zone.today` (or the equivalent variant).
2. It checks for usage of `.to_time`.

The check against `Date.today` is great, and largely fool-proof - if it detects it, it flags it as an issue.  Problem solved.  In our code base, we want to mark these as errors and faill the build.  Great.

However, `.to_time` unfortunately isn't as iron-clad, and is producing a number of false-positives in the codebase.

Doing this raises an error with rubocop with `Rails/Date` enabled:
```ruby
start = Time.zone.now
next_time = start.to_time
```
This might be extremely useful for some people, and that's great.  However it would be great if those were controlled by separate levers in the configuration file.  Possibly something like `Rails/ToTime`

## Steps to reproduce the problem
1. Enable `Rails/Date` cop
2. Enter the following:
```ruby
a = Time.zone.today
b = a.to_time
```
This is extremely important! Providing us with a reliable way to reproduce
a problem will expedite its solution.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.53.0 (using Parser 2.5.0.3, running on ruby 2.2.4 x86_64-darwin16)
```

**Comment by btoconnor** [comment]
Note: I am aware that I am not on the latest rubocop release, but after going through the changelog I did not notice anything relating to `Rails/Date`.  Please let me know if you'd like me to upgrade before discussing further.
